File: Java/151-reverse-words-in-a-string.java
//151. Reverse Words in a String

//Given an input string s, reverse the order of the words.
//Example 2:

//Input: s = "  hello world  "
//Output: "world hello"
//Explanation: Your reversed string should not contain leading or trailing spaces.
    
package leetcode;

class Solution {

    public static String removeSpaces(String input) {
        return input.trim(); 
    }

    public String reverseWords(String s) {
 
        s = removeSpaces(s);

        String[] words = s.split("\\s+"); 

        StringBuilder reversed = new StringBuilder();
        for (int i = words.length - 1; i >= 0; i--) {
            reversed.append(words[i]);
            if (i > 0) {
                reversed.append(" "); 
            }
        }

        return reversed.toString();
    }

    public static void main(String[] args) {
        Solution solution = new Solution();
        String result = solution.reverseWords("  the sky is blue  ");
        System.out.println("Resultado: \"" + result + "\""); 
    }
}

File: Python/151-reverse-words-in-a-string.py
class Solution:

    @staticmethod
    def remove_spaces(input_string):
        return input_string.strip() 

    def reverse_words(self, s):
        s = self.remove_spaces(s)
        words = s.split() 
        reversed_words = " ".join(reversed(words))
        return reversed_words

solution = Solution()
result = solution.reverse_words("  the sky is blue  ")
print(f"Resultado: \"{result}\"") 
